### PR TITLE
sqlotus backend

### PR DESCRIPTION
**Pull request by willscott**
incorporate the blockstore prototyped at https://github.com/ribasushi/ltsh/commit/7d564a08bbd

### CODE REVIEW

File: lib/tstracker/tracker.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	}
+
+	tsk, err := types.TipSetKeyFromBytes(tskBytes)
+	if err != nil {
+		return nil, -1, err
+	}
+
+	return &tsk, epoch, nil
+}
+
```
Comment (ribasushi) [inline lib/tstracker/tracker.go:33]: Really the only thing you need from this file is this function, but doesn't hurt to c/p

**Review by ribasushi — APPROVED** [review]
LGTM - I should have the pg ready for query around 22 UTC

File: lib/tstracker/tracker.go [hunk 2]
```diff
@@ -31,6 +31,14 @@ func (tcs *tcs) CurrentDbTipSetKey(ctx context.Context) (*types.TipSetKey, abi.C
...
 }
 
+func (tcs *tcs) GetCurrentTipset(ctx context.Context) []cid.Cid {
+	tsk, _, err := tcs.CurrentDbTipSetKey(ctx)
+	if err != nil {
+		return []cid.Cid{}
+	}
+	return tsk.Cids()
+}
+
```
Comment (ribasushi) [inline lib/tstracker/tracker.go:41]: Huh... @willscott are you sure you need this? The tsk is what you get by feeding these CID's to the tipset maker no...?

Comment (willscott) [reply lib/tstracker/tracker.go:41]: that's the function statediff is actually using 😁

Comment (ribasushi) [reply lib/tstracker/tracker.go:41]: Ok, I will make a method like this then in the lib, it's pretty silly to do that twice ;)

Comment (willscott) [reply lib/tstracker/tracker.go:41]: don't worry about it. i can wrap it in my data source thing also. other libraries should be looking for more useful errors, which is what your current interface has. mine is just sloppy about it.

**Pull request merged**